internal/indexmanagement/scripts.go

```diff
@@ -131,6 +131,83 @@ def updateWriteIndex(currentIndex, nextIndex, alias):
     print(e)
     sys.stdout = original_stdout
     return False
+
+def getMaxAllowedSize(diskThreshold):
+  # Returns the total disk space of all nodes combined multiplied by the pre-defined threshold
+  original_stdout = sys.stdout
+  try:
+    es_client = getEsClient()
+    response = es_client.cat.allocation(format="JSON", h="disk.total", bytes="kb")
+    totalSize = 0
+    for i in range(len(response)):
+      totalSize = totalSize + int(response[i]["disk.total"])
+    return int(diskThreshold/100.0 * totalSize)
+  except Exception as e:
+    sys.stdout = open('/tmp/response.txt', 'w')
+    print(e)
+    sys.stdout = original_stdout
+    return -1
+
+def getDeletableIndices(index, alias, maxAllowedSize, indicesSizeCounter, indicesToDelete):
+  # Returns a list of indices from [alias] (infra, app, or audit), that should be deleted
+  # traverse through indices from newest to oldest, adding up their size. When the sum of sizes exceeds
+  # maxAllowedSize, start adding indices to the list
+  original_stdout = sys.stdout
+  try:
+    expectedSize = int(index['store.size']) + indicesSizeCounter
+    if indicesSizeCounter < maxAllowedSize and expectedSize < maxAllowedSize:
+      indicesSizeCounter += int(index['store.size'])
```

Comment:
Maybe this instead? Is there ever a time where `expectedSize` < `indicesSizeCounter`?

```suggestion
    if expectedSize < maxAllowedSize:
      indicesSizeCounter += int(index['store.size'])
```

Reply:
Thanks @Red-GV
I have updated the code with you suggestions + merged two functions into one (the ones that return the list of indices to delete)